core/src/main/scala/bekindrewind/VcrClient.scala

```diff
@@ -29,7 +30,18 @@ trait VcrClient {
     records
   }
 
-  val newlyRecorded: AtomicReference[Vector[VcrRecord]] =
+  def addNewRecord(recordRequest: VcrRecord): Unit =
+    newlyRecorded.updateAndGet { records =>
+      val transformed = recordRequest.copy(
+        request = recordOptions.requestTransformer(recordRequest.request),
+        response = recordOptions.responseTransformer(recordRequest.response)
+      )
+      records :+ transformed
+    }
+
+  def currentNewlyRecorded(): immutable.Seq[VcrRecord] = newlyRecorded.get()
```

Comment:
I understand the meaning behind it, but this feels like an awkward name to me because "current" and "new" are contradictory. Maybe `getNewlyRecorded` instead? Or just `newlyRecorded` and then rename the `AtomicReference` one `newlyRecordedRef` or something like that? Just an idea.